If hw_base_encode_send_frame() is called with frame == NULL before any
input frames are submitted, ctx->pic_end is NULL and dereferencing it
causes a crash. This can occur in hardware encoding pipelines that flush
before receiving any input frames.

Fix by checking ctx->pic_end before accessing its pts field.

Link: https://github.com/tvheadend/tvheadend/issues/1833
Signed-off-by: James Hutchinson <jahutchinson99@googlemail.com>

Source: https://patchwork.ffmpeg.org/project/ffmpeg/patch/20250719070836.283597-1-jahutchinson99@googlemail.com/

--- libavcodec/hw_base_encode.c-orig	2025-11-21 01:15:18.000000000 +0000
+++ libavcodec/hw_base_encode.c	2025-12-27 00:27:24.108168862 +0000
@@ -503,7 +503,7 @@ static int hw_base_encode_send_frame(AVC
 
         // Fix timestamps if we hit end-of-stream before the initial decode
         // delay has elapsed.
-        if (ctx->input_order <= ctx->decode_delay)
+        if (ctx->input_order <= ctx->decode_delay && ctx->pic_end)
             ctx->dts_pts_diff = ctx->pic_end->pts - ctx->first_pts;
     }
 
